File: code_swift_v2/45.swift
/* 
 * 圆圈中最后剩下的数字
 */
func lastRemaining(_ n: Int, _ m: Int) -> Int {
    if n == 1 {
        return 0
    }
    
    var pos: Int = 0
    for i in 2 ..< n+1 {
        pos = (pos + m) % i
    }
    return pos
}

/***** 测试部分 *****/
let ans = lastRemaining(8, 3)
print(ans)
/******************/

File: code_cpp_v1/45.cpp
/*
 * 圆圈中最后剩下的数字
 */
int lastRemaining(int n, int m)
{
    if (n == 1)
        return 0;
    int pos = 0;
    for (int i = 2; i <= n; i++)
        pos = (pos + m) % i;
    return pos;
}
